docs/source/design.rst

```diff
@@ -1,7 +1,151 @@
-.. _internals:
+.. _design:
 
-*********************
-Ibis design internals
-*********************
+Design
+======
 
-More to come here.
+Primary Goals
+*************
+
+#. Type safety
+#. Expressiveness
+#. Composability
+#. Familiarty
+
+Flow of execution
+-----------------
+
+#. User writes expression
+#. Each method or function call builds a new expression
+#. Expressions are type checked as you create them
+#. Some expressions have some optimizations that happen as the user builds them
+   (EXAMPLE)
+#. Some optimizations happen at compile time? (EXAMPLE)
+#. Expressions are compiled
+#. The SQL string that generated by the compiler is sent to the database and
+   executed
+#. The database returns some data that is then turned into a pandas DataFrame
+   by ibis
+
+
+Expressions
+-----------
+
+The main user-facing component of ibis is expressions. The base class of all
+expressions in ibis is the :class:`~ibis.expr.types.Expr` class.
+
+Expressions provide the user facing API, defined in ``ibis/expr/api.py``
+
+Type System
+~~~~~~~~~~~
+
+Ibis's type system consists of a set of rules for specifying the types of
+inputs to nodes. Upon construction of a Node subclass, ibis performs validation
+of every input to the node based on the rule that was used to declare the
+input.
+
+Rules are defined in ``ibis/expr/rules.py``
+
+The ``Expr`` class
+~~~~~~~~~~~~~~~~~~
+
+Expressions are a thin but important abstraction over operations, containing
+only type information and shape information, i.e., whether they are tables,
+columns, or scalars.
+
+Examples of expressions include :class:`~ibis.expr.types.Int64Column`,
+:class:`~ibis.expr.types.StringScalar`, and
```

Comment:
show an example of an expr, maybe use Log and Add (as you describe below)

``(1 + t.column).log()``

Reply:
Done